dhcpv4/bsdp/client.go

```diff
@@ -3,22 +3,23 @@
 package bsdp
 
 import (
-	"fmt"
-	"net"
-	"syscall"
+	"errors"
 
 	"github.com/insomniacslk/dhcp/dhcpv4"
 )
 
-// Client is a BSDP-specific client suitable for performing BSDP exchanges.
-type Client dhcpv4.Client
-
 // Exchange runs a full BSDP exchange (Inform[list], Ack, Inform[select],
 // Ack). Returns a list of DHCPv4 structures representing the exchange.
-func (c *Client) Exchange(ifname string, informList *dhcpv4.DHCPv4) ([]dhcpv4.DHCPv4, error) {
+func Exchange(client *dhcpv4.Client, ifname string, informList *dhcpv4.DHCPv4) ([]dhcpv4.DHCPv4, error) {
```

Comment:
why changing this to a function? (same thing below for `dhcpv4.go`)

I'd rather have separate `dhcpv4` and `bsdp` clients. They are also technically two different clients, that may talk with different servers

Reply:
I couldn't come up with a good way to share the implementation of `SendReceive` unless this became a function. Having a separate `bsdp.Client` struct made it so that I couldn't use the implementation of `SendReceive`, which does all the hard work of setting up a connection to send broadcast packets.

Is there a better way to organize this so that I only have to effectively override the `Exchange` function?